_posts/2020/2020-12-10-how-to-handle-class-names.md

```diff
@@ -0,0 +1,141 @@
+
+```
+.component-name {}
+.component-name--modifier-name {}
+.component-name__descendant-name {}
+```
+
+{% image_custom image="/assets/img/pages/blog/images/class-names-example.png" width="90" %}
+
+#### BEM code example
+```
+// HTML structure
+<div class="author-card author-card--full-size">
+  <img class="author-card__image author-card__image--placeholder" src="img/image.jpg">
+  <p class="author-card__description">
+    <strong class="author-card__name">Author name</strong> 
+    lorem ipsum dolor...
+  </p>
+</div>
+
+// LESS nesting
+.author-card {
+  &__image {
+    &--placeholder {}
+  }
+  &__description {}
+  &__name {}
+  &--full-size {}
+}
+```
+
+
+### SUIT CSS
+
+Suit CSS focuses on supporting especially the UI component-based web application development e.g. React, Ember, and Angular.
+Regarding the [SUIT CSS naming convention](https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md){:target="_blank"} we use double hyphens to indicate a component modifier e.g. `.AuthorCard--example`, and a single hyphen to indicate a component descendant e.g. `.AuthorCard-image`. Components, modifiers, and descendants names are written in camelcase, and component names start with a capital. 
+
+But now we get to the interesting part: Nicolas Gallagher added a different type of class called "helper" or "utility" class. You can add this class on any element within a component to provide low-level structural or positional traits, beginning with the prefix of `u-` e.g. `.u-fullSize`. Another addition is to have states classes which are reflecting changes to a component state, they always begin with `is-` e.g. `.is-empty`.
+
+```
+.u-utilityName {}
+
+.ComponentName {}
+.ComponentName--modifierName {}
+.ComponentName-descendentName {}
+
+.ComponentName.is-stateOfComponent
+```
+
+#### SUIT CSS code example
+```
+// HTML structure
+<div class="AuthorCard u-fullSize">
+  <img class="AuthorCard-image is-empty" src="img/image.jpg">
+  <p class="AuthorCard-description">
+    <strong class="AuthorCard-author">Author name</strong> 
+    lorem ipsum dolor...
+  </p>
+</div>
+
+// LESS nesting
+.u-fullSize {}
+.AuthorCard {
+  &-image {
+    &.is-empty {}
+  }
+  &-description {}
+  &-name {}
+}
+```
+
+
+## Summery
+
+As you can see, the declaration of the classes is so well defined that it takes the work off your shoulders to think about class names in detail. But there are a lot of other advantages, why you should consider using **BEM** or **SUIT CSS** for your projects. 
+
+**Pros:**
+* Clear and predefined naming conventions makes understanding as well as debugging much easier
+* Flat CSS structure makes specificity easier
+* Using unique and context-related class names avoids CSS conflicts
+* Composing and reusing blocks reduces the amount of maintainable CSS code
+* Scalability apply to projects of all sizes
+* **BEM:** More popular naming convention for classes leads to more public Q&A
```

Comment:
I don't understand "more public Q&A"

Reply:
...= "BEM is better known and you can find more about it on the internet." 😬
(I wrote something else -> In the end it should become clear that BEM has advantages because it is much better known!)